packages/buidler-etherscan/src/solc/bytecode.ts

```diff
@@ -0,0 +1,260 @@
+import { RunTaskFunction } from "@nomiclabs/buidler/types";
+
+import { metadataLengthSize, readSolcMetadataLength } from "./metadata";
+import { InferralType } from "./version";
+
+export async function lookupMatchingBytecode(
+  contractFiles: CompilerOutput["contracts"],
+  deployedBytecode: string,
+  inferralType: InferralType
+) {
+  for (const [contractFilename, contracts] of Object.entries(contractFiles)) {
+    for (const [contractName, contract] of Object.entries(contracts)) {
+      // Normalize deployed bytecode according to this contract.
+      const { deployedBytecode: runtimeBytecodeSymbols } = contract.evm;
+
+      const comparison = await compareBytecode(
+        deployedBytecode,
+        runtimeBytecodeSymbols,
+        inferralType
+      );
+
+      if (comparison.match) {
+        const {
+          contractInformation: {
+            immutableValues,
+            libraryLinks,
+            normalizedBytecode,
+          },
+        } = comparison;
+        // The bytecode matches
+        return {
+          immutableValues,
+          libraryLinks,
+          normalizedBytecode,
+          contractFilename,
+          contractName,
+          contract,
+        };
+      }
+    }
+  }
+
+  return null;
+}
+
+type BytecodeComparison =
+  | { match: false }
+  | { match: true; contractInformation: BytecodeExtractedData };
+
+interface BytecodeExtractedData {
+  immutableValues: ImmutableValues;
+  libraryLinks: ResolvedLinks;
+  normalizedBytecode: string;
+}
+
+export async function compareBytecode(
+  deployedBytecode: string,
+  runtimeBytecodeSymbols: CompilerOutputBytecode,
+  inferralType: InferralType
+): Promise<BytecodeComparison> {
+  let bytecodeSize = deployedBytecode.length;
+  // We will ignore metadata information when comparing. Etherscan seems to do the same.
+  if (inferralType !== InferralType.METADATA_ABSENT) {
+    // The runtime object may contain nonhexadecimal characters due to link placeholders.
```

Comment:
If I'm reading this correctly, this comment just says that this will break

Reply:
I got it. I agree that this requires a comment, but the current comment doesn't explain it.